### PR TITLE
Refactored code to improve control and error handling

### PR DESCRIPTION
**Pull request by xZyzz7**
Refactored code in `generate_config()` by dividing it into two smaller functions `config_options()` and `config_push()` for improved readability and error handling.

Made some changes to `verify_dir()` and `overwrite_confirmation()` by removing `if not os.path.exists(directory):`  keeping this check in `generate_config()` is better to give us more control over the functions for error handling

### CODE REVIEW

**Review by LyubomirT — CHANGES_REQUESTED** [review]
If possible, could you please only apply error handling? The new code looks a bit overcomplicated and seemingly would be better if left as it is in terms of structure.

**Comment by xZyzz7** [comment]
The new code may look a bit overcomplicated because of a lot of refactoring but in reality i didn't change anything really just then main function `generate_config()` was doing a lot so i substracted some of its instruction into two functions. Tell me what part of the code look complicated i'll happily explain.

**Comment by LyubomirT** [comment]
> The new code may look a bit overcomplicated because of a lot of refactoring but in reality i didn't change anything really just then main function `generate_config()` was doing a lot so i substracted some of its instruction into two functions. Tell me what part of the code look complicated i'll happily explain.

Personally, I don't have any issues understanding the code, but I'm worried about future contributors. However, considering that the config generator won't be actively modified, there should be no big issues with that. Thank you for your contribution! :)

I'll review the changes the second time real quick, and will merge right after that.

**Review by LyubomirT — APPROVED** [review]

**Pull request merged**
